routemaster/models/client_token.rb

```diff
@@ -0,0 +1,37 @@
+require 'routemaster/models'
+require 'routemaster/mixins/redis'
+require 'securerandom'
+
+module Routemaster
+  module Models
+    class ClientToken
+      include Mixins::Redis
+
+      PREFIX = "api_keys:"
+
+      def self.get_all
+        key_dump = {}
+        existing_keys = _redis.keys "#{PREFIX}*"
```

Comment:
The thing is that KEYS is proportional to the number of keys in Redis, not to the number of keys that match. So if a subscriber is backed up it could be a large number.